web/src/engine/websites/ShonenMagazine.ts

```diff
@@ -0,0 +1,38 @@
+import { Tags } from '../Tags';
+import icon from './ShonenMagazine.webp';
+import { DecoratableMangaScraper, type MangaPlugin, Manga } from '../providers/MangaPlugin';
+import * as CoreView from './decorators/CoreView';
+import * as Common from './decorators/Common';
+import { FetchCSS, FetchRequest } from '../FetchProvider';
+
+@Common.MangaCSS(/^https?:\/\/pocket\.shonenmagazine\.com\/episode\/\d+$/, CoreView.queryMangaTitleFromURI)
+@CoreView.ChaptersSinglePageCSS(CoreView.queryChapters)
+@CoreView.PagesSinglePageJSON()
+@CoreView.ImageAjax()
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        //Shonenmagazine.com && pocket.shonenmagazine.com mangas url starts with https://pocket.shonenmagazine.com so this plugin handles both.
+        super('shonenmagazine', `週刊少年マガジ (Weekly Shonen Magazine & Pocket Magazine)`, 'https://pocket.shonenmagazine.com', Tags.Language.Japanese, Tags.Source.Official, Tags.Media.Manga);
+    }
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+
+        //fetch from pocket.shonenmagazine.com [/series]
+        const mangas = await CoreView.FetchMangasMultiPageCSS.call(this, provider, ['/series'], 'div.series-items ul.daily-series > li.daily-series-item > a');
```

Comment:
A shorter, but dirty solution would be to temporary replace the `URI`
```typescript
this.URI.href = 'https://pocket.shonenmagazine.com'
const mangas1 = await CoreView.FetchMangasMultiPageCSS.call(this, provider, ['/series'], 'div.series-items ul.daily-series > li.daily-series-item > a');

this.URI.href = 'https://shonenmagazine.com'
const mangas2 = await CoreView.FetchMangasMultiPageCSS.call(this, provider, ['/series/smaga', '/series/bmaga', '/series/others']);
return [ ...mangas1, ...mangas2 ];
```

Reply:
Because its dirty i didnt want to do it, knowing you will be reviewing :)
I can do it we just have to make sure the url is the pocket one at the end. Gotta try catch this to be safe.